fastn-core/src/library2022/processor/figma_tokens.rs

```
@@ -0,0 +1,267 @@
+            color_title = color_title,
+            color_list = json_string_dark_values,
+        );
+    }
+
+    let json_formatted_light = final_light;
+    let json_formatted_dark = final_dark;
+
+    let full_cs = format!(
+        "{{\n\"{}-light\": {{\n{}\n}},\n\"{}-dark\": {{\n{}\n}} \n}}",
+        variable_name.as_str(),
+        json_formatted_light,
+        variable_name.as_str(),
+        json_formatted_dark
+    );
+
+    let response_json: serde_json::Value = serde_json::Value::String(full_cs);
+    doc.from_json(&response_json, &kind, line_number)
+}
+
+pub fn capitalize_word(s: &str) -> String {
+    let mut c = s.chars();
+    match c.next() {
+        None => String::new(),
+        Some(f) => f.to_uppercase().collect::<String>() + c.as_str(),
+    }
+}
+
+fn extract_light_dark_colors<'a>(
+    value: ftd::ast::VariableValue,
+    doc: &mut ftd::interpreter2::TDoc<'a>,
+    variable_name: &mut String,
+    light_colors: &mut ftd::Map<ftd::Map<VT>>,
+    dark_colors: &mut ftd::Map<ftd::Map<VT>>,
+    line_number: usize,
+) -> ftd::interpreter2::Result<()> {
+    if let ftd::ast::VariableValue::Record { headers, .. } = &value {
+        let header = headers.get_by_key_optional("variable", doc.name, line_number)?;
+        if let Some(variable) = header {
+            if let ftd::ast::VariableValue::String { value: hval, .. } = &variable.value {
+                *variable_name = hval.trim_start_matches('$').to_string();
+                let bag_entry = doc.resolve_name(hval);
+                let bag_thing = doc.bag().get(bag_entry.as_str());
+
+                if let Some(ftd::interpreter2::Thing::Variable(v)) = bag_thing {
+                    let property_value = &v.value;
+
+                    if let ftd::interpreter2::PropertyValue::Value {
+                        value: ftd::interpreter2::Value::Record { fields, .. },
+                        ..
+                    } = property_value
+                    {
```

Throw error instead of just ignoring it. Also, use an early exit policy.